packages/libs/react-components/README.md

```diff
@@ -73,3 +70,72 @@ You can start storybook after installing:
 ```
 rushx storybook
 ```
+
+## Component Library Guidelines
+
+We would like to maintain a strict component library with opinionated styles and
+isolated components that rely on wrapper or layout components for positioning.
+The following is some information about how we have approached building our
+components to maintain this goal.
+
+### Styling
+
+We are currently using Stitches as our CSS solution although we are looking into
+other zero-runtime options like
+[vanilla-extract](https://vanilla-extract.style/) and
+[Tailwind CSS](https://tailwindcss.com/).
+
+_Theming_
+
+We have defined a theme using elements of the Kadena Design System and these
+tokens should be used as property values in most cases to ensure consistency and
+alignment with the design. With stitches, we are also able to override this
+theme within projects to add additional tokens or update colors for a dark
+theme, for example.
+
+_Colors_
+
+In our theme we have color scales that represent a set of different shades for a
+color as well as theme specific tokens that return different colors depending on
+whether we are in light/dark theme.
+
+Due to the strict nature of our component library, we should use our disgression
+to determine what subset of colors should be exposed for a certain component.
+For example, a text component would likely only be used in a small subset of
+colors on an interface - one approach to doing this would be to create a color
```

Comment:
should it be `one approach`?
or was this already decided? (not sure anymore)

Reply:
ah fair. I do think we decided this was the best way to do this. I'll replace "One approach" to "our recommended approach"